packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -602,83 +538,123 @@ export class NftDetectionController extends StaticIntervalPollingController<
         .address;
+
+    const { selectedNetworkClientId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    const {
+      configuration: { chainId },
+    } = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      selectedNetworkClientId,
+    );
+
     /* istanbul ignore if */
-    if (!this.isMainnet() || this.#disabled) {
+    if (!supportedNftDetectionNetworks.includes(chainId) || this.#disabled) {
       return;
     }
     /* istanbul ignore else */
     if (!userAddress) {
       return;
     }
 
-    const apiNfts = await this.#getOwnerNfts(userAddress);
-    const addNftPromises = apiNfts.map(async (nft) => {
-      const {
-        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        tokenId: token_id,
-        contract,
-        kind,
-        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        image: image_url,
-        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        imageSmall: image_thumbnail_url,
-        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        metadata: { imageOriginal: image_original_url } = {},
-        name,
-        description,
-        attributes,
-        topBid,
-        lastSale,
-        rarityRank,
-        rarityScore,
-        collection,
-      } = nft.token;
-
-      let ignored;
-      /* istanbul ignore else */
-      const { ignoredNfts } = this.#getNftState();
-      if (ignoredNfts.length > 0) {
-        ignored = ignoredNfts.find((c) => {
-          /* istanbul ignore next */
-          return (
-            c.address === toChecksumHexAddress(contract) &&
-            c.tokenId === token_id
-          );
-        });
-      }
-
-      /* istanbul ignore else */
-      if (!ignored) {
-        /* istanbul ignore next */
-        const nftMetadata: NftMetadata = Object.assign(
-          {},
-          { name },
-          description && { description },
-          image_url && { image: image_url },
-          image_thumbnail_url && { imageThumbnail: image_thumbnail_url },
-          image_original_url && { imageOriginal: image_original_url },
-          kind && { standard: kind.toUpperCase() },
-          lastSale && { lastSale },
-          attributes && { attributes },
-          topBid && { topBid },
-          rarityRank && { rarityRank },
-          rarityScore && { rarityScore },
-          collection && { collection },
-        );
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    const updateKey: `${Hex}:${string}` = `${chainId}:${userAddress}`;
```

Comment:
Similar issue: https://github.com/MetaMask/eth-snap-keyring/issues/323#issuecomment-2160760043

This eslint-disable directive should be removable with the next `@metamask/keyring-api` release.

- Currently, `userAddress` is resolving to `any`. 
- This is because the `AccountsController:getSelectedAccount` action returns `InternalAccount`,
- which is defined using `superstruct` instead of `@metamask/superstruct` in current the `@metamask/keyring-api` version.
- https://github.com/MetaMask/superstruct/pull/1